pyart/correct/bias_and_noise.py

```diff
@@ -0,0 +1,155 @@
+    Radar, JAOT, 23, 1340-1356
+
+    """
+    # parse the field parameters
+    if urhohv_field is None:
+        urhohv_field = get_field_name('uncorrected_cross_correlation_ratio')
+    if snr_field is None:
+        snr_field = get_field_name('signal_to_noise_ratio')
+    if zdr_field is None:
+        zdr_field = get_field_name('differential_reflectivity')
+    if nh_field is None:
+        nh_field = get_field_name('noisedBZ_hh')
+    if nv_field is None:
+        nv_field = get_field_name('noisedBZ_vv')
+    if rhohv_field is None:
+        rhohv_field = get_field_name('cross_correlation_ratio')
+
+    # extract fields from radar
+    if urhohv_field in radar.fields:
+        urhohv = radar.fields[urhohv_field]['data']
+    else:
+        raise KeyError('Field not available: ' + urhohv_field)
+    if snr_field in radar.fields:
+        snrdB_h = radar.fields[snr_field]['data']
+    else:
+        raise KeyError('Field not available: ' + snr_field)
+    if zdr_field in radar.fields:
+        zdrdB = radar.fields[zdr_field]['data']
+    else:
+        raise KeyError('Field not available: ' + zdr_field)
+    if nh_field in radar.fields:
+        nh = radar.fields[nh_field]['data']
+    else:
+        raise KeyError('Field not available: ' + nh_field)
+    if nv_field in radar.fields:
+        nv = radar.fields[nv_field]['data']
+    else:
+        raise KeyError('Field not available: ' + nv_field)
+
+    snr_h = np.ma.power(10., 0.1*snrdB_h)
+    zdr = np.ma.power(10., 0.1*zdrdB)
+    alpha = np.ma.power(10., 0.1*(nh-nv))
+
+    mask = np.ma.getmaskarray(urhohv)
+    fill_value = urhohv.get_fill_value()
+
+    rhohv_data = urhohv*np.ma.sqrt((1.+1./snr_h)*(1.+zdr/(alpha*snr_h)))
+    rhohv_data[rhohv_data > 1.] = 1.
+    rhohv_data.set_fill_value(fill_value)
+    rhohv_data.data[mask.nonzero()] = fill_value
+
+    rhohv = get_metadata(rhohv_field)
+    rhohv['data'] = rhohv_data
+
+    return rhohv
+
+
+def correct_bias(radar, bias=0., field_name=None):
+    """
+    Corrects a radar data bias. If field name is none the correction is
+    applied to horizontal reflectivity by default
+
+    Parameters
+    ----------
+    radar : Radar
+        radar object
+
+    bias : float
+        the bias magnitude
+
+    field_name: str
+        names of the field to be corrected
+
+    Returns
+    -------
+    corrected_field : dict
+        The corrected field
+
+    """
+    # parse the field parameters
+    if field_name is None:
+        field_name = get_field_name('reflectivity')
+
+    # extract fields from radar
+    if field_name in radar.fields:
+        field_data = radar.fields[field_name]['data']
+    else:
+        raise KeyError('Field not available: ' + field_name)
+
+    mask = np.ma.getmaskarray(field_data)
+    fill_value = field_data.get_fill_value()
+
+    corr_field_data = field_data - bias
+    corr_field_data.set_fill_value(fill_value)
+    corr_field_data.data[mask] = fill_value
```

Comment:
`corr_field_data.data[mask] = np.ma.masked` and remove the line which sets the fill_value.